initializr-web/src/main/java/io/spring/initializr/web/project/ProjectRequestToDescriptionConverter.java

```diff
@@ -68,16 +69,57 @@ public ProjectDescription convert(ProjectRequest request,
 		description.setPlatformVersion(Version.parse(springBootVersion));
 		description.setVersion(determineValue(request.getVersion(),
 				() -> metadata.getVersion().getContent()));
+		conventionCorrection(description);
```

Comment:
We can clean the value when we set it itself like we do in `description.setPackageName(getPackageName(request, metadata));`